boards/xtensa/m5stack_core2/m5stack_core2.dts

```diff
@@ -0,0 +1,254 @@
+/*
+ * Copyright (c) 2023 Martin Kiepfer <mrmarteng@teleschirm.org>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32/esp32_common.dtsi>
+#include "m5stack_core2-pinctrl.dtsi"
+#include "grove_connectors.dtsi"
+#include <zephyr/dt-bindings/display/ili9xxx.h>
+#include <zephyr/dt-bindings/regulator/axp192.h>
+
+/ {
+	model = "esp32";
+	compatible = "espressif,esp32";
+
+	aliases {
+		pwr-led = &pwr_led;
+		uart-0 = &uart0;
+		i2c-0 = &i2c0;
+		watchdog0 = &wdt0;
+		rtc = &pfc8563_rtc;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+		zephyr,display = &ili9342c;
+		zephyr,keyboard-scan = &kscan_input;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,rtc = &pfc8563_rtc;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&cpu1 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&psram0 {
+	reg = <0x3f800000 DT_SIZE_M(8)>;
+	status = "disabled";
+};
+
+&uart0 {
+	status = "okay";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart0_tx_gpio1 &uart0_rx_gpio3>;
+	pinctrl-names = "default";
+};
+
+&uart1 {
+	status = "disabled";
+	current-speed = <115200>;
+	pinctrl-0 = <&uart1_rx_gpio33 &uart1_tx_gpio32>;
+	pinctrl-names = "default";
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 21 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 22 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_default>;
+	scl-timeout-us = <0>;
+	pinctrl-names = "default";
+
+	pfc8563_rtc: pfc8563@51 {
+		compatible = "nxp,pcf8523";
+		reg = <0x51>;
+		status = "okay";
+		battery-switch-over = "disabled";
+	};
+
+	axp192_pmic: axp192@34 {
+		compatible = "x-powers,axp192";
+		reg = <0x34>;
+		status = "okay";
+
+		axp192_regulator: axp192_regulator {
+			compatible = "x-powers,axp192-regulator";
+			status = "okay";
+
+			vdd_mcu: DCDC1 {
+				regulator-init-microvolt = <3350000>;
+				regulator-min-microvolt = <3200000>;
+				regulator-max-microvolt = <3400000>;
+				regulator-initial-mode = <AXP192_DCDC_MODE_AUTO>;
+				regulator-boot-on;
+				regulator-always-on;
+			};
+
+			lcd_bg: DCDC3 {
+				regulator-init-microvolt = <2800000>;
+				regulator-min-microvolt = <0>;
+				regulator-max-microvolt = <2800000>;
+				regulator-boot-on;
```

Comment:
not sure we want the LCD always on? Isn't the fact that a vin-supply is defined on the display node already enough to bind the status of the regulator to the status of the display? @gmarull

Reply:
I am actually no longer sure if the vin-supply binding to lcd_bg regulator is correct. Is vin-supply intended to control the regulator for the control logic? ili9xx driver is not supporting brightness control. Hence I guess the vin-supply setting does not have any effect. Only a few, very specific display driver (e.v. max7219) support backlight control yet.